### PR TITLE
Material UI theming

### PR DESCRIPTION
**Pull request by mxc-maggiechen**
## Notion Ticket Link
[Material UI Theming](https://www.notion.so/uwblueprintexecs/Material-UI-Theming-1478fb2a170b43708521ea44e43ed527?pvs=4)

## Implementation Description
- **Added `theme.ts`**: This file includes the theme settings for Material UI, following the [Figma design file](https://www.figma.com/design/83NY0r7MI8Y2r14YWeMnzE/Material-3-Design-Kit-(Extend-a-Family)?node-id=49823-12141&t=i8BBHtmwjvJ6vnKE-0). It currently contains:
  - Custom theme overrides for easier developer navigation. For example, there are now sets of colors named "learner," "administrator," and "facilitator".
  - Maps of colors and fonts specified in Figma, centralizing any future adjustments to these themes in this file.
  - A `getTheme(bool)` function that returns a theme object with either the dark or light theme color palette based on the passed boolean, facilitating easy dev testing.

- **Added `palette.ts`**: This file includes maps of the tonal palettes from the [Figma design file](https://www.figma.com/design/83NY0r7MI8Y2r14YWeMnzE/Material-3-Design-Kit-(Extend-a-Family)?node-id=49823-12141&t=i8BBHtmwjvJ6vnKE-0), simplifying the process of adjusting color palettes within the theme.

## Steps to Test

**Testing Colors:**

**Use Case 1:**
1. Go to a page and add `const theme = useTheme()`.
2. Set the color of a visible component (e.g., text or button) using `style={{ color:  theme.palette.learner.main }}}`.

**Use Case 2:**
1. Create a visible Material UI component on a page.
2. Set the color property, e.g., `<Button color="learner.main"></Button>`.

**Testing Fonts:**
1. Create a visible Material UI Typography component on a page.
2. Set the variant property, e.g., `<Typography variant="displayLarge">Maggie Chen</Typography>`.

## Focus Areas for Reviewers
- Please review the custom theme overrides and ensure they align with the design specifications.
- Verify the functionality of the `getTheme` function for both dark and light themes.
- Confirm the ease of use and correctness of color and font mapping as per the Figma design.

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/theme/palette.ts
Comment (98ZhaoJeffrey) [inline frontend/src/theme/palette.ts:1]: i think using string as the type for colors isn't the best approach and we can just define a color type like in this stackoverflow thread:
https://stackoverflow.com/questions/42584228/how-can-i-define-a-type-for-a-css-color-in-typescript

File: frontend/src/theme/theme.ts [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    labelMedium?: TypographyStyleOptions;
+    labelSmall?: TypographyStyleOptions;
+    bodyLarge?: TypographyStyleOptions;
+    bodyMedium?: TypographyStyleOptions;
+    bodySmall?: TypographyStyleOptions;
+  }
+
+  interface Theme {
+    status: {
+      danger: string;
```
Comment (98ZhaoJeffrey) [inline frontend/src/theme/theme.ts:46]: i assume that status is something like "warning" | "success" | "error", so maybe we can replace string with a custom status type instead

**Review by 98ZhaoJeffrey — COMMENTED** [review]
looks fine to me, just two small comments

**Comment by carolynzhang18** [comment]
beautiful PR description, brought a tear to my eye 🥹 really appreciate the detail there

File: frontend/src/theme/theme.ts [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    letterSacing: "0pt",
+  },
+  titleMedium: {
+    fontSize: "16pt",
+    fontWeight: 600,
+    lineHeight: "24pt",
+    letterSacing: "+0.15pt",
+  },
+  titleSmall: {
+    fontSize: "24pt",
```
Comment (carolynzhang18) [inline frontend/src/theme/theme.ts:181]: 14? hehe

File: frontend/src/theme/theme.ts [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    dark: error[10],
+  },
+};
+
+const typography: TypographyOptions = {
+  displayLarge: {
+    fontSize: "57pt",
+    fontWeight: 700,
+    lineHeight: "64pt",
+    letterSacing: "-0.25pt",
```
Comment (carolynzhang18) [inline frontend/src/theme/theme.ts:136]: letterSpacing?

**Review by carolynzhang18 — COMMENTED** [review]
thanks for working through all this MUI setup!! super well done, just some minor things and then we good!
image

**Review by carolynzhang18 — APPROVED** [review]

**Pull request merged**